### DISCUSSION TITLE
Thoughts around Sandstorm Packaging

### DISCUSSION THREAD

**Discussion by brianmarshall** [firefly-iii/firefly-iii#6893] [General]
Hello,

I'm running the last packaged version of Firefly-III on a self-hosted Sandstorm.io instance, and it is working a treat.
I couldn't find any discussion on here around it, but I assume that the sandstorm.io company failing was the reason why Firefly III hasn't been packaged for a long time. 
Now that a community has rallied around sandstorm to keep it viable over the long term, is there any way that packaging could be resumed to allow current and future users to stay on the latest version? Maybe a way to automate them as part of your build pipeline? Some way I can help?
If I'm wrong and there are other fundamental issues, please let me know!
Thanks!

   -Brian

**Comment by JC5** [comment]
Hey Brian,

I stopped support at the time when Sandstorm went down, yes. 

Packaging can resume of course, that's not a question at all. But I'm not going to do it. I just don't have the time to manage separate packages for all of the self hosting services there are out there.

**Reply by brianmarshall** [reply]
Hi James, thanks for the reply!
I'm willing to do the packaging if I'm able to continue it in a reasonable fashion. Any chance you have notes around how you were doing it before?
The main problem I see is that (as far as I can tell) unless I sign the releases with the same key that was used previously it won't allow existing installs to upgrade and will instead show up like a new package.
Any other potential problems I haven't considered? Problems with table/data formats etc with the big version jump?
Any thoughts or ideas?
Thanks!

  -Brian

**Reply by JC5** [reply]
At the time, there was a script called `vagrant-spk` which did all the work. The last build that contains Sandstorm related code is this one: https://github.com/firefly-iii/firefly-iii/releases/tag/4.7.5.2

If I remember correctly, (i no longer have the necessary scripts) you could build Firefly III fairly easily with this. But I no longer have the original instructions.

I still have the original key pair although it's long since expired. If you get it working, let me know, and we can see if I can hand over the key, or use it to sign a new one.

**Reply by brianmarshall** [reply]
Thanks for being willing to help!
My sandstorm shows the last version at 4.7.9, so I'll try to build/package that as a known-good starting point and figure out how to transition to current once that's working
I'll let you know how I make out
